lib/vagrant-libvirt/driver.rb

```diff
@@ -239,6 +239,7 @@ def get_ipaddress_from_qemu_agent(domain, machine_id, timeout)
 
         unless addresses.nil?
           addresses['return'].each do |interface|
+            next unless interface.key?('hardware-address')
             next unless domain.mac.downcase == interface['hardware-address'].downcase
```

Comment:
could probably solve by switching to using `interface.fetch('hardware-address', '').downcase`, but I'd have to check if there is a scenario where `domain.mac` is an empty string as well, so probably better to start with this for now.